classes/Product.php

```diff
@@ -7833,24 +7833,20 @@ public function setAdvancedStockManagement($value)
     }
 
     /**
-     * Get the default category according to the shop.
+     * Get the default category id according to the shop.
      *
-     * @return array{id_category_default: int}|int
+     * @return int
      */
-    public function getDefaultCategory()
+    public function getDefaultCategory(): int
     {
-        $default_category = Db::getInstance()->getValue(
+        $defaultCategory = Db::getInstance()->getValue(
             'SELECT product_shop.`id_category_default`
             FROM `' . _DB_PREFIX_ . 'product` p
             ' . Shop::addSqlAssociation('product', 'p') . '
             WHERE p.`id_product` = ' . (int) $this->id
         );
 
-        if (!$default_category) {
-            return ['id_category_default' => Context::getContext()->shop->id_category];
-        } else {
-            return (int) $default_category;
-        }
+        return (int) $defaultCategory ?? Context::getContext()->shop->id_category;
```

Comment:
```suggestion
        return (int) ($defaultCategory ?? Context::getContext()->shop->id_category);
```

Without the brackets the casting to int is applied to `$defaultCategory`. If `$defaultCategory` is null it will return `0` instead of `Context::getContext()->shop->id_category`.
https://3v4l.org/gIpZB